# Notebook 8 — Business Recommendations
**Purpose:** Synthesize findings from Notebooks 1-7 into a ranked set of
business recommendations with quantified, clearly-labeled *simulated* impact
estimates, feeding directly into the Business Consulting Report and CEO deck.

**Libraries used:** `pandas`, `matplotlib`


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

DATA_DIR = "../01_Data"
sales = pd.read_csv(f"{DATA_DIR}/sales_clean.csv", parse_dates=["Transaction_Date"])
inventory = pd.read_csv(f"{DATA_DIR}/inventory.csv")

monthly = sales.set_index('Transaction_Date').resample('MS').agg(
    Revenue=('Sales_Amount','sum'), Profit=('Profit','sum')
)
monthly['Margin_Pct'] = 100*monthly['Profit']/monthly['Revenue']

first_q_margin = monthly['Margin_Pct'].iloc[:3].mean()
latest_q_margin = monthly['Margin_Pct'].iloc[-3:].mean()
annualized_revenue = sales['Sales_Amount'].sum() / 2   # 2 years of data

margin_gap = first_q_margin - latest_q_margin
recovery_opportunity = annualized_revenue * (margin_gap/100)

print(f"Margin, first quarter observed : {first_q_margin:.1f}%")
print(f"Margin, latest quarter observed: {latest_q_margin:.1f}%")
print(f"Margin gap                     : {margin_gap:.1f} points")
print(f"Approx. annualized revenue base: NGN {annualized_revenue:,.0f}")
print(f"Estimated annual profitability recovery opportunity if margin gap is closed:")
print(f"   NGN {recovery_opportunity:,.0f}  (SIMULATED — directional sizing only)")


Margin, first quarter observed : 23.2%
Margin, latest quarter observed: 20.1%
Margin gap                     : 3.2 points
Approx. annualized revenue base: NGN 505,744,182
Estimated annual profitability recovery opportunity if margin gap is closed:
   NGN 16,167,532  (SIMULATED — directional sizing only)


In [ ]:
slow_moving = inventory[inventory['Stock_Level'] > inventory['Reorder_Level']*2]
overstock_value = slow_moving['Inventory_Cost'].sum()
target_reduction_pct = 0.18
working_capital_release = overstock_value * target_reduction_pct

print(f"Inventory value tied up in slow-moving stock: NGN {overstock_value:,.0f}")
print(f"Target reduction (18%): NGN {working_capital_release:,.0f} working-capital release (SIMULATED)")

stockout_risk = inventory[inventory['Stock_Level'] <= inventory['Reorder_Level']]
print(f"\nProduct-store combinations at/below reorder level: {len(stockout_risk)} of {len(inventory)} "
      f"({100*len(stockout_risk)/len(inventory):.0f}%)")


Inventory value tied up in slow-moving stock: NGN 47,950,312
Target reduction (18%): NGN 8,631,056 working-capital release (SIMULATED)

Product-store combinations at/below reorder level: 305 of 874 (35%)


## Prioritized Recommendations (feeding the Business Consulting Report)

| # | Recommendation | Basis | Simulated Impact |
|---|---|---|---|
| R1 | Introduce discount-approval thresholds; cap non-seasonal discounts at 20% | Notebook 5 & 7 — margin falls sharply above 15-30% discount bands | ~NGN 15-16M/year profitability recovery |
| R2 | Rationalize slow-moving SKUs and renegotiate or delist bottom-margin products | Notebook 5 — bottom 15 products by margin identified | Frees working capital, improves category margin mix |
| R3 | Reduce overstock in flagged product-store combinations | Notebook 6/8 — ~10% of inventory records are overstocked | ~NGN 8.6M working-capital release, ~18% holding-cost reduction |
| R4 | Fix replenishment cadence at branches with stockout risk | Notebook 8 — ~35% of product-store combinations at/below reorder level | Reduced lost-sale risk, improved service level |
| R5 | Centralize reporting and add Campaign_ID to POS to close the sales-marketing data gap | Section 2/7 — campaigns cannot currently be tied to POS transactions | Enables true marketing ROI attribution |
| R6 | Launch a structured retention program for high-CLV customers | Notebook 4 — top 5% of customers contribute a disproportionate share of revenue | Protects highest-leverage revenue base |
| R7 | Re-score `Customer_Value_Category` on a recurring cadence | Notebook 4 — label broadly tracks spend but needs periodic refresh | Keeps targeting/segmentation accurate over time |

**Next:** These recommendations, together with the quantified findings from
Notebooks 3-7, are formalized in `05_Reports/Business_Consulting_Report.docx`
and presented to leadership in `06_Presentation/ApexMart_CEO_Presentation.pptx`.
